packages/medusa/src/models/order-edit.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+} from "typeorm"
+
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import { resolveDbType } from "../utils/db-aware-column"
+import { OrderItemChange } from "./order-item-change"
+import { SoftDeletableEntity } from "../interfaces"
+import { generateEntityId } from "../utils"
+import { LineItem } from "./line-item"
+import { Order } from "./order"
+import { User } from "./user"
+
+@FeatureFlagEntity("order_editing")
+export class OrderEdit extends SoftDeletableEntity {
+  @Column({ nullable: true })
+  order_id: string
+
+  @ManyToOne(() => Order, (o) => o.edits)
+  @JoinColumn({ name: "order_id" })
+  order: Order
+
+  @OneToMany(() => OrderItemChange, (oic) => oic.order_edit, {
+    eager: true,
+    cascade: true,
+  })
+  changes: OrderItemChange[]
+
+  @Column({ nullable: true })
+  internal_note?: string
+
+  @Column({ nullable: true })
+  created_by: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
```

Comment:
suggestion: I think it would make sense to allow the data model to support order edits that are customer-created. 

E.g. imagine that you have a customer account universe where you can view your orders and create an edit. This should have "created_by: [customer id]" instead of admin user. 

I.e. -> Remove the foreign key and make the created_by field allow both customer id and user id.